docs/cookbook/shared-state.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Shared State
+description: How to use a reactive composable to share your objects across component files.
+author: whitespacecode
+thumbnail: /recipes/animations.png
+difficulty: 0
+---
+
+# Shared State in TresJS
+
+This guide will help you get started with shared state in TresJS by building a simple scene with a cube that can be shared across component files.
+
+<StackBlitzEmbed project-id="tresjs-minimal-reproduction-rycc4j" />
+
+## Creating a State Composable
+
+First, we'll create a composable to store the objects.
+
+### Setting the object to the state
+Next, we'll assign an object to the state and include it in a subcomponent where we can access and use it.
+
+### Using the object in other components
+With the mesh assigned to the reactive state, it's available throughout your project.
+
+::: code-group
+
+```ts [composables/state.ts]
+import { reactive, toRefs } from "vue";
+
+const state = reactive({
+    mesh: null,
+    //you can add more objects here
+})
+
+export function useState() {
+    return {
+        ...toRefs(state)
```

Comment:
We should slim down the reactivity here.

Feel free to point out if I'm misunderstanding, but I believe `Ref<state.mesh>` will be deeply reactive. Since a single attached THREE.Mesh contains the entire THREE.Scene graph (+ materials, geometries, and so on), that normally creates *a lot* of reactivity that goes unused.

It's fine if users want to use `Ref<Mesh>` in their own code, but we have run into concrete performance problems in the past even in relatively modest scenes. So as an official recipe, I think it would be best to refactor. 

Maybe one or both of:

* use `shallow*` alternatives
* use slimmer data, i.e. `state.isMeshVisible = true`

@alvarosabu Thoughts here?

Reply:
That's correct, better to use shallowRefs or shallowReactive. I wonder if `toRefs` automatically use `shallowRef` it when the state is `shallowReactive`  🤔

Reply:
>  I wonder if toRefs automatically use shallowRef it when the state is shallowReactive

In the Vue docs, the type indicates that `Ref` values are returned from `toRefs`.

https://vuejs.org/api/reactivity-utilities.html#torefs

Reply:
Yes please feel free to change the .md file.
Most of the code is stuff i gathered from discord or by using the [stackblitz of alvarosabu](https://stackblitz.com/@alvarosabu/collections/tresjs)